File: app/src/at-nhatnguyen/java/asiantech/internship/summer/javacore/Exercise25.java
package asiantech.internship.summer.javacore;
public class Exercise25 {
    private static boolean isPrimeNumber(int n){
        if(n>1){
            for(int i=2;i<=Math.sqrt(n);i++){
                if(n%i==0) return false;
            }
            return true;
        }
        else return false;
    }
    private static boolean isReversibleNumber(int n){
        StringBuilder x= new StringBuilder();
        String str= ""+n;
        x.append(str);
        String check= "" + x.reverse();
        return str.equals(check);
    }
    private static boolean primeNumber(int n){
        while(n!=0){
            if(!isPrimeNumber(n%10)) return false;
            n/= 10;
        }
        return true;
    }
    public static void main(String[] args) {
        int i,count= 0;
        System.out.println("cac so tu 5-7 chu so thoa man dieu kien la: ");
        for(i=22223 ; i<7777777 ; i+=2){
            if(isPrimeNumber(i) && primeNumber(i) && isReversibleNumber(i)){
                System.out.println(" "+i); count++;
            }
        }
        System.out.println(" Co "+count+" so thoa man");
    }
}

File: app/src/at-nhatnguyen/java/asiantech/internship/summer/kotlin/Exercise25.kt
package asiantech.internship.summer.kotlin

private fun isReversibleNumber(n: Int): Boolean {
    val x = StringBuilder()
    val str = n.toString()
    x.append(str)
    val check = x.reverse().toString()
    return str == check
}

private fun primeNumber(n: Int): Boolean {
    var a: Int = n
    while (a != 0) {
        if (!isPrimeNumber(a % 10)) {
            return false
        }
        a /= 10
    }
    return true
}

fun main() {
    var i = 22223
    var count = 0
    println("cac so tu 5-7 chu so thoa man dieu kien la: ")
    while (i < 7777777) {
        if (isPrimeNumber(i) && primeNumber(i) && isReversibleNumber(i)) {
            println(" $i")
            count++
        }
        i += 2
    }
    println(" Co $count so thoa man")
}
